include/filters.php

```diff
@@ -235,6 +241,33 @@ public function get_pages( $pages, $args ) {
 		return $pages;
 	}
 
+	/**
+	 * Filters the WP_Query in get_pages() per language.
+	 *
+	 * @since 3.4
+	 *
+	 * @param array $query_args  Array of arguments passed to WP_Query.
+	 * @param array $parsed_args Array of get_pages() arguments.
+	 * @return array Array of arguments passed to WP_Query with the language.
+	 */
+	public function get_pages_query_args( $query_args, $parsed_args ) {
+		if ( isset( $parsed_args['lang'] ) && '' === $parsed_args['lang'] ) {
+			$query_args['lang'] = '';
+			return $query_args;
+		}
+
+		$language = empty( $parsed_args['lang'] ) ? $this->curlang : $this->model->get_language( $parsed_args['lang'] );
+
+		if ( empty( $language ) || ! $this->model->is_translated_post_type( $parsed_args['post_type'] ) ) {
+			$query_args['lang'] = '';
+			return $query_args;
+		}
+
+		$query_args['lang'] = $language->slug;
```

Comment:
This is far too complex.
- There is no need to make a specific case for `'lang' => ''`.
- `WP_Query` will check for translated post type iself as it won't handle the parameter if the query is not registered.
- The current language is already handled by our `WP_Query` filter
- If you get a `PLL_Language` object here, that's a bug and it should be fixed in [the right place](https://github.com/polylang/polylang/blob/3.4.2/admin/admin-classic-editor.php#L334) to receive only a language slug.

```suggestion
		if ( isset( $parsed_args['lang'] ) ) {
			$query_args['lang'] = $parsed_args['lang'];
		}
```